strum_macros/src/enum_iter.rs

```diff
@@ -62,14 +62,24 @@ pub fn enum_iter_inner(ast: &syn::DeriveInput) -> TokenStream {
         #[allow(missing_docs)]
         #vis struct #iter_name #ty_generics {
             idx: usize,
+            back_idx: usize,
             marker: ::std::marker::PhantomData #phantom_data,
         }
 
+        impl #impl_generics #iter_name #ty_generics #where_clause {
+            fn get(&self, idx: usize) -> Option<#name #ty_generics> {
+                match idx {
+                    #(#arms),*
+                }
+            }
+        }
+
         impl #impl_generics ::strum::IntoEnumIterator for #name #ty_generics #where_clause {
             type Iterator = #iter_name #ty_generics;
             fn iter() -> #iter_name #ty_generics {
                 #iter_name {
-                    idx:0,
+                    idx: 0,
+                    back_idx: 0,
```

Comment:
I might set `back_idx` to `#variant_count - 1` so that one counts down and the other counts up.

Reply:
I tried this at first but got an underflow exception on the last element. This approach seemed like the easiest fix.